From 1e2aefa84d4b3e04b838fc5464812e6dd94061f9 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:05 -0400
Subject: [fs] nfs: Use error handler on failed GETATTR with successful OPEN

Message-id: <1403798077-28799-47-git-send-email-steved@redhat.com>
Patchwork-id: 84949
O-Subject: [RHEL7.1 PATCH 46/78] NFSv4: Use error handler on failed GETATTR with successful OPEN
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Andy Adamson <andros@netapp.com>

Place the call to resend the failed GETATTR under the error handler so that
when appropriate, the GETATTR is retried more than once.

The server can fail the GETATTR op in the OPEN compound with a recoverable
error such as NFS4ERR_DELAY. In the case of an O_EXCL open, the server has
created the file, so a retrans of the OPEN call will fail with NFS4ERR_EXIST.

Signed-off-by: Andy Adamson <andros@netapp.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 8935ef664ea33a7ddd39015b26fe35aa7744f715
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4proc.c b/fs/nfs/nfs4proc.c
index 44fb93a..68dd81e 100644
--- a/fs/nfs/nfs4proc.c
+++ b/fs/nfs/nfs4proc.c
@@ -2027,7 +2027,7 @@ static int _nfs4_proc_open(struct nfs4_opendata *data)
    return status;
  }
  if (!(o_res->f_attr->valid & NFS_ATTR_FATTR))
-  _nfs4_proc_getattr(server, &o_res->fh, o_res->f_attr, o_res->f_label);
+  nfs4_proc_getattr(server, &o_res->fh, o_res->f_attr, o_res->f_label);
  return 0;
 }
 
-- 
1.7.1